### PR TITLE
Fix sidenav links

### PR DESCRIPTION
**Pull request by sknep**
## TL;DR: Fixes Staging links
This pull request uses the repo name to inform the root-relative urls in the sidenav and across the site, so that links on Staging will point to Staging and not Production. It also now shows the current/active page in the left sidenav.

### This pull request changes...
- the default `site.baseurl` has been changed to the staging repo name, which is only really used for local development. GitHub Pages ignores this value when deciding where to host the site, using the repo name as the base path instead. Unfortunately, it still generates all the interior pages using the original `site.baseurl` value, which is how you can end up with a repo hosted at `/X/...` and all `site.baseurl`-prepended links with `/Y/...`.
- For links to pages and assets across the site, the URLs now use the repo name (`site.github.repository_name`) rather than using `site.baseurl`, ensuring that no matter which repo the build runs on, it uses the correct path. This value comes from GitHub during the build, so it should be always in sync with the deployment environment.
- Sidenav links now show whether they're the current page (or the parent of a current page).

**Please note that should we start to host this on a subdomain, the references to `site.github.repository_name` as well as `site.baseurl` should be removed.** Also, it would be necessary to have two subdomains, one for prod and one for staging.

**Steps to manually review and verify this change...**
1. Unfortunately, this is the kind of change that has to be tested by deploying. Please contact me if you have any questions, and I'd like to be present for the merges both to staging and to main.

### This pull request can be merged when…
- [x] The above pull request description is complete
- [ ] The pull request author has tested the change successfully
- [x] The pull request author has assigned a CMS reviewer
- [x] The change has been reviewed and approved by CMS

#### Notes
- https://jekyll.github.io/github-metadata/site.github/
- http://jekyllrb.com/docs/github-pages/
- https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/

### CODE REVIEW

**Review by nickarrow — APPROVED** [review]
Thanks much, this is a great catch and fix.
